colossalai/tensor/colo_tensor.py

```diff
@@ -147,87 +45,27 @@ def is_model_data(self) -> bool:
     def __torch_function__(cls, func, types, args=(), kwargs=None):
         global _COLOSSAL_OPS
         if func in _COLOSSAL_OPS:
-            for arg in args:
-                if isinstance(arg, ColoTensor):
-                    return _COLOSSAL_OPS[func](types, args, kwargs, None)
-
-            for kwarg in kwargs.values():
-                if isinstance(kwarg, ColoTensor):
-                    return _COLOSSAL_OPS[func](types, args, kwargs, None)
-        else:
-            # If we have not hijact the function, convert the ColoTensors in args and kwargs to torch tensors.
-            args = [arg.torch_tensor() if isinstance(arg, ColoTensor) else arg for arg in args]
-            if kwargs is None:
-                kwargs = {}
-
-            kwargs = {k: v.torch_tensor() if isinstance(v, ColoTensor) else v for k, v in kwargs.items()}
-            return cls._filter_outputs_with_colo(func(*args, **kwargs))
-
-    def backward(self, gradient: Optional[torch.Tensor] = None, retain_graph: bool = False):
-        self._torch_tensor.backward(gradient=gradient, retain_graph=retain_graph)
-
-    def __add__(self, o) -> "ColoTensor":
-        if isinstance(o, ColoTensor):
-            return ColoTensor.init_from_torch_tensor(self.torch_tensor() + o.torch_tensor())
-        elif isinstance(o, (torch.Tensor, int, float)):
-            return ColoTensor.init_from_torch_tensor(self.torch_tensor() + o)
-        else:
-            raise TypeError(f'{type(o)} is not supported in ColoTensor __add__')
-
-    __radd__ = __add__
-
-    def __truediv__(self, o) -> "ColoTensor":
-        return ColoTensor.init_from_torch_tensor(self.torch_tensor() / o)
-
-    def __getattr__(self, name):
-
-        def replace_tensor_with_colo(func):
+            func = _COLOSSAL_OPS[func]
+        return super().__torch_function__(func, types, args, kwargs)
```

Comment:
confusing about this line? why not return _COLOSSAL_OPS[func](types, args, kwargs, None) as before?

Reply:
`super().__torch_function__` handles type conversion automatically and disable torch function in this context. See https://github.com/pytorch/pytorch/blob/0e351c7df9c41367bca94b824e20f68ea8f165b4/torch/_tensor.py#L1112

Reply:
As ColoTensor inherits torch.Tensor, we must disable __torch_function__ in our implementations of torch functions. This avoids infinite recursion.

Reply:
I see. You are smart.